internal/controller/etcdopstask/utils.go

```diff
@@ -0,0 +1,308 @@
+func (r *Reconciler) handleRequeue(ctx context.Context, logger logr.Logger, task *druidv1alpha1.EtcdOpsTask, result handler.Result, phase druidv1alpha1.OperationPhase) ctrlutils.ReconcileStepResult {
+	var opType druidv1alpha1.LastOperationType
+
+	if phase == druidv1alpha1.OperationPhaseCleanup {
+		opType = druidv1alpha1.LastOperationTypeDelete
+	} else {
+		opType = druidv1alpha1.LastOperationTypeReconcile
+	}
+
+	statusUpdate := taskStatusUpdate{
+		Operation: &druidv1alpha1.LastOperation{
+			Type:        opType,
+			State:       druidv1alpha1.LastOperationStateRequeue,
+			Description: result.Description,
+		},
+		Error: result.Error,
+	}
+
+	if err := r.updateTaskStatus(ctx, task, statusUpdate); err != nil {
+		if result.Error != nil {
+			return ctrlutils.ReconcileWithError(errors.Wrapf(result.Error, "failed to record error: %v", err))
+		}
+		return ctrlutils.ReconcileWithError(err)
+	}
+
+	if result.Error != nil {
+		return ctrlutils.ReconcileWithError(result.Error)
+	}
+
+	message := fmt.Sprintf("Task %s in progress", strings.ToLower(string(phase)))
+	logger.Info("Requeuing task", "phase", phase, "requeueInterval", r.config.RequeueInterval.Duration)
+	return ctrlutils.ReconcileAfter(r.config.RequeueInterval.Duration, message)
+}
+
+// handleError handles error scenarios with phase-specific behavior
+func (r *Reconciler) handleError(ctx context.Context, logger logr.Logger, task *druidv1alpha1.EtcdOpsTask, result handler.Result, phase druidv1alpha1.OperationPhase) ctrlutils.ReconcileStepResult {
+	var taskState *druidv1alpha1.TaskState
+	var errorMessage string
+	var opType druidv1alpha1.LastOperationType
+
+	switch phase {
+	case druidv1alpha1.OperationPhaseAdmit:
+		taskState = ptr.To(druidv1alpha1.TaskStateRejected)
+		errorMessage = "task rejected, handing to deletion flow"
+		opType = druidv1alpha1.LastOperationTypeReconcile
+	case druidv1alpha1.OperationPhaseRunning:
+		taskState = ptr.To(druidv1alpha1.TaskStateFailed)
+		errorMessage = "task completed"
+		opType = druidv1alpha1.LastOperationTypeReconcile
+	case druidv1alpha1.OperationPhaseCleanup:
+		errorMessage = "cleanup failed"
+		opType = druidv1alpha1.LastOperationTypeDelete
+	}
+
+	statusUpdate := taskStatusUpdate{
+		Operation: &druidv1alpha1.LastOperation{
+			Type:        opType,
+			State:       druidv1alpha1.LastOperationStateError,
+			Description: result.Description,
+		},
+		State: taskState,
+		Error: result.Error,
+	}
+
+	if err := r.updateTaskStatus(ctx, task, statusUpdate); err != nil {
+		return ctrlutils.ReconcileWithError(errors.Wrapf(result.Error, "failed to record error: %v", err))
+	}
+
+	if phase == druidv1alpha1.OperationPhaseAdmit || phase == druidv1alpha1.OperationPhaseRunning {
+		logger.Info("Task cleanup after TTL", "state", *task.Status.State, "ttlSeconds", task.Spec.TTLSecondsAfterFinished)
+		return ctrlutils.ReconcileAfter(task.GetTimeToExpiry(), fmt.Sprintf("Task %s, waiting for TTL to expire", strings.ToLower(string(*task.Status.State))))
```

Comment:
did not understand this logic. If the phase is `Admit` or `Running` and there is an error then why should it be requeued after timetoexpiry?